addons/mail/models/mail_blacklist.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class MailBlackList(models.Model):
+    """ Model of blacklisted email addresses to stop sending emails."""
+    _name = 'mail.blacklist'
+    _inherit = ['mail.thread']
+    _description = 'Mail Blacklist'
+    _rec_name = 'email'
+
+    email = fields.Char(string='Email Address', required=True, index=True, help='This field is case insensitive.',
+                        track_visibility='always')
+    active = fields.Boolean(default=True, track_visibility='onchange')
+
+    _sql_constraints = [
+        ('unique_email', 'unique (email)', 'Email address already exists!')
+    ]
+
+    @api.model_create_multi
+    def create(self, values):
+        """ To avoid crash during import due to unique email, return the existing records if any """
+        sql = '''SELECT id, email FROM mail_blacklist
+                WHERE LOWER(email) = any (array[%s])
+                ''' % (', '.join(["'" + value['email'].lower() + "'" for value in values]))
+        self._cr.execute(sql)
+        records = self._cr.fetchall()
+
+        blacklisted_emails = [rec[1] for rec in records]
+        blacklisted_ids = [rec[0] for rec in records]
+        non_blacklisted_records = [value for value in values if value['email'] not in blacklisted_emails]
+
+        results = super(MailBlackList, self).create(non_blacklisted_records)
+        return self.env['mail.blacklist'].browse(blacklisted_ids + results.ids)
```

Comment:
return self.browse(bl_ids) | results

To check: I don't know if create multi preserve order. In which case you would probably have to reorder your returned record set according to incoming order.